src/dxvk/dxvk_presenter.cpp

```diff
@@ -48,6 +57,7 @@ namespace dxvk {
 
 
   VkResult Presenter::acquireNextImage(PresenterSync& sync, uint32_t& index) {
+    std::lock_guard<dxvk::mutex> lock(m_lowLatencyMutex);
```

Comment:
AcquireNextImage can block. I think holding a mutex for the entire duration of acquire can cause potential trouble.

Reply:
I see the same concern was also raised in vkd3d-proton review, so I suppose this can be addressed in a similar way (only hold the lock in case of recreation).

Reply:
With the locking in acquire and present removed my layer-based LL2 implementation works with significantly less stutters — so this is probably that has to be fixed. I think the locks aren't really necessary here anyway since access to swapchain is externally synchronized (which implies it can't be destroyed for the duration of call).

Reply:
The lock should only be taken when the swapchain is destroyed now. So I think this issue can be considered resolved.